### ISSUE TITLE
[Feature Request]: Make initial set up more user friendly

### ISSUE THREAD

**Issue by kadisonm** [sboesen/remotely-sync#49]
### What feature are you suggesting?

As mentioned in #44 

It might be worth prompting the user to authenticate and intial sync before displaying additional sync settings. This could make the initial set up less daunting for new users.

Alternatively, a modal that appears on plugin install explaining the process. Or an update to the read me. 
Worth discussing.

### Ensure no sensitive information

- [X] I ensure that no sensitive information is submitted in the issue.

**Comment by kadisonm** [comment]
The current process is definitely quite cluttered and confusing at first to me. I think a guided installation or clearer seperation of the settings could help.

In the case of a guided installation, a button to skip could be implemented to instantly get access to the full settings. 

I'd imagine it would be guided in these steps with settings being revealed after each one:
1. Authenticate with a remote service.
2. Set an encryption password or leave empty. (Make clear to press confirm to reveal next stage)
3. Propt an initial sync. (With a model and a button, make the user wait until the vault is synced, then tell the user to configure rest of settings)
4. Configure sync settings and additional settings.

**Comment by kadisonm** [comment]
@MartinJDavis93
> Until the problem can be reproduced, there is no point in such radical implementations. Well, that's my opinion.

I do agree with this but at the very least I think a general process should be more clear to first time users.

**Comment by MartinJDavis93** [comment]
> @MartinJDavis93
> 
> > Until the problem can be reproduced, there is no point in such radical implementations. Well, that's my opinion.
> 
> I do agree with this but at the very least I think a general process should be more clear to first time users.

This all may be cool, but right now there are more pressing issues to address. I could also come up with a dozen different improvements, but I don't see the point in cluttering the issues at this stage of plugin development. It'll only distract the author. If you think we can do without some innovations, most likely that's the case.

In comparison with other plugins, especially those related to Key File or Token for authentication, or the need to register a device, for example, through the Google Actions Console, the settings of this plugin seem incredibly simple and just great.

**Comment by kadisonm** [comment]
> This all may be cool, but right now there are more pressing issues to address. 

I agree but it isn't bad to discuss it still.

Explaining my thoughts with words sometimes doesn't express well, so I prefer to show it visually. 

I think setting an encryption password first would be better and then displaying the auth settings. Then the rest could just be a guide under the auth modal. However, I'm not sure exactly how this would go for webdav or s3.

![Guide](https://github.com/sboesen/remotely-secure/assets/134670047/94ca6c29-f01d-4911-9a62-5691116d1a80)

**Comment by kadisonm** [comment]
Additionally, the second modal can be removed and this would just be a nice update to the auth modals.

**Comment by MartinJDavis93** [comment]
I agree that it's possible to make the plugin interface more user-friendly in principle. However, this doesn't negate the fact that during further plugin configuration, the user will have to deal with all these parameters.

As for starting with "Encryption Password", it's not the best option, considering that the goal is to improve the interface to make it more user-friendly. I understand the practicality of it, but it'll not look very welcoming. It would definitely throw me off, especially considering that I don't use encryption. I understand that this step, well, I hope 😄, can be skipped for those who don't use encryption, but still, it's not ideal.

It would probably be better to have a welcome page first, on which you can choose or start the "Basic Configuration" of the plugin (for example, in the form of a big button), and below in simple text, "I'm an experienced user, I'll configure everything myself at once" 😅. If you choose an "Experienced User", it redirects you straight to the plugin, as it does now. If you choose "Basic Configuration", on the next page, ask the question: "Do you want to use encryption?" And depending on the answer, display the page with "Encryption Settings" or proceed to "Authentication".

All the inscriptions I mentioned are conditional, just to convey the essence.

**Comment by kadisonm** [comment]
> It would definitely throw me off

Yeah I agree. It would be best as an additional setting after but I'm not entirely sure what happens if you add an encryption password after syncing without one. If it's not necessary for the password to be set before syncing then definitely have it after.

> "I'm an experienced user, I'll configure everything myself at once" 😅

Definitely.

**Comment by sboesen** [comment]
> > It would definitely throw me off
> 
> 
> 
> Yeah I agree. It would be best as an additional setting after but I'm not entirely sure what happens if you add an encryption password after syncing without one. If it's not necessary for the password to be set before syncing then definitely have it after.
> 
> 
> 
> > "I'm an experienced user, I'll configure everything myself at once" 😅
> 
> 
> 
> Definitely.

It complains and won't sync, I believe. Good to configure what you want up front or work having to wipe cloud storage to start over

**Comment by kadisonm** [comment]
> It complains and won't sync, I believe. Good to configure what you want up front or work having to wipe cloud storage to start over

Yeh that was my understanding, but I hadn't tested it. On a side note, there should probably be somewhere telling the user to back up their vault then wipe their cloud if they are changing the password. Or offer a feature to do that for them and remove their auth (after double checking they backed up the vault)

**Comment by sboesen** [comment]
I don't think I want to add that feature (delete it for them), foot-guns and all, but agreed on calling it out in an alert or something.

**Comment by sboesen** [comment]
I agree we should have a section in the readme, but it's already long as is and I suspect many people aren't reading it as is. A lot of the readme can probably be moved elsewhere & slimmed down, but I think it's worth exploring how other plugins have done it if it doesn't add too much code complexity to Remotely Secure.

This is definitely way down on the priority list (for me at least), lots of other stuff including refactoring the code coming first.

**Comment by kadisonm** [comment]
> This is definitely way down on the priority list (for me at least), lots of other stuff including refactoring the code coming first.

Of course, this is definitely future quality of life improvements.

I found most plugins handle settings and such the same except excalidraw. Which I think does things in a really creative user friendly way.

![image](https://github.com/sboesen/remotely-secure/assets/134670047/df81400f-83bc-42e6-bbd2-fa40fa68daf9)
